### PR TITLE
Port of #10760 in preview-1

### PR DESCRIPTION
**Pull request by ViktorHofer**
Port of https://github.com/dotnet/sdk/pull/10760

### CODE REVIEW

**Review by nguerrera — APPROVED** [review]
Change is good, I don't know if we can take this in P1 or not. Please work with @dsplaisted and @mmitche  to figure that out.  It seems in the same vein as the other code flow issue that has lots of things out of date in P1 that are being looked at

**Comment by mmitche** [comment]
> Change is good, I don't know if we can take this in P1 or not. Please work with @dsplaisted and @mmitche to figure that out. It seems in the same vein as the other code flow issue that has lots of things out of date in P1 that are being looked at

You can. Go for it.

**Comment by ViktorHofer** [comment]
@nguerrera who can help diagnosing the failing tests?

File: eng/Versions.props [hunk 1]
```diff
@@ -42,7 +39,7 @@
   </PropertyGroup>
   <PropertyGroup>
     <!-- Dependencies from https://github.com/nuget/nuget.client -->
-    <NuGetBuildTasksPackageVersion>5.5.0-preview.2.6355</NuGetBuildTasksPackageVersion>
```
Comment (jaredpar) [inline eng/Versions.props]: Was the problem before simply that we had multiple entries for `<NuGetBuildTasksPackageVersion>` in the Version.props file? DARC was then simply updating the first one but not the second hence we did not get any true dependency flow here?

If so can we please file a bug in arcade. This seems like a really easy case to catch and given it's likely to contribute to a servicing event it's one we should definitely consider implementing.

**Pull request merged**